packages/angular/ssr/test/npm_package/BUILD.bazel

```diff
@@ -8,28 +9,50 @@ ts_library(
     srcs = glob(["**/*.ts"]),
     deps = [
         "@npm//@bazel/runfiles",
-        "@npm//@types/diff",
     ],
 )
 
 jasmine_node_test(
     name = "test",
     srcs = [":unit_test_lib"],
     data = [
-        "THIRD_PARTY_LICENSES.txt.golden",
         "//packages/angular/ssr:npm_package",
-        "@npm//diff",
     ],
 )
 
-nodejs_binary(
-    name = "test.accept",
-    testonly = True,
-    data = [
-        "THIRD_PARTY_LICENSES.txt.golden",
-        ":unit_test_lib",
+genrule(
+    name = "critters_license_file",
+    srcs = [
         "//packages/angular/ssr:npm_package",
-        "@npm//diff",
     ],
-    entry_point = ":update-package-golden.ts",
+    outs = [
+        "THIRD_PARTY_LICENSES.txt",
+    ],
+    cmd = """
+      cp $(location //packages/angular/ssr:npm_package)/third_party/critters/THIRD_PARTY_LICENSES.txt $(location :THIRD_PARTY_LICENSES.txt)
+    """,
+)
+
+diff_test(
+    name = "critters_license_test",
+    failure_message = """
+
+    To accept the new golden file, execute:
+    yarn bazel run //packages/angular/ssr/test/npm_package:critters_license_test.accept
+    """,
+    file1 = ":THIRD_PARTY_LICENSES.txt.golden",
+    file2 = ":critters_license_file",
+)
+
+write_file(
+    name = "critters_license_test.accept",
+    out = "critters_license_file_accept.sh",
+    content =
+        [
+            "#!/usr/bin/env bash",
```

Comment:
Neat trick with `write_file`, didn't know it could create a binary like that.

One limitation here is that this will only work for Unix developers, not Windows devs. It's probably not a huge deal, especially since anyone could manually copy over the text, even if that is a bit annoying.

Maybe once we get on `@aspect_rules_js` it might be worth making a more reusable and cross-platform `golden_test` macro. I thought `@aspect_bazel_lib` had something here, but it seems like [their `diff_test`](https://docs.aspect.build/rulesets/aspect_bazel_lib/docs/diff_test/) isn't much smarter than Skylib's.

Reply:
We could create a macro that creates a CMD instead a bash for Windows.

But as mentioned there are workaround, and likely the update of the golden will be done by one of team which at the moment use Linux or Mac.